src/daemon/rpm-ostreed-automatic.service

```diff
@@ -5,4 +5,5 @@ ConditionPathExists=/run/ostree-booted
 
 [Service]
 Type=simple
+Nice=10
```

Comment:
Unfortunately this won't do what you want here, because ultimately this just calls into `rpm-ostreed.service`.

I think by default one would need to use systemd drop-ins for `rpm-ostreed.service` locally to have the desired effect.

In theory perhaps though, we could better differentiate "background updates" from "foreground activity" and the daemon could automatically lower its priority.

Hmm.  Possibly there's a way we could inherit niceness and other properties from this unit.

Reply:
Wow, you're quick to reply :open_mouth: 
Isn't the systemd service the right place to configure this? The service has a key made for this. Configuring "niceness" in the process itself is extra complexity, which would probably just end up calling nicectl from within anyway...

Reply:
I moved the niceness key to the daemon. When would you want rpm-ostreed to run at 0 niceness? It's such a heavy command, that I would always like it to be nice. Running the command manually or when automatic updates are applied, my system stutters.

Reply:
Does this really help you though?

I suspect the main problem is not CPU usage but I/O bandwidth for most people.

Reply:
Indeed, for the most part. There are still CPU bound parts

Writing OSTree commit: rpm-ostree start-daemon is CPU intensive (1-2 cores, i guess)
When I tried adding a new package, `rpm-ostree start-daemon` was using ~180% CPU.

Then, `Nice` is maybe not the right solution. I just want my system to run smoothly when the updates randomly start.

Reply:
Or what.... man pages suggest that setting `nice` for the process also makes the I/O nice
https://linux.die.net/man/1/ionice